blazor/datagrid/foreignkey-column.md

```diff
@@ -0,0 +1,768 @@
+            {
+                int code = 10;
+                for (int i = 1; i < 2; i++)
+                {
+                    Orders.Add(new OrderData(10248,1, "Reims", 32.18));
+                    Orders.Add(new OrderData(10249,2, "Münster",33.33));
+                    Orders.Add(new OrderData(10250,3, "Rio de Janeiro",12.35));
+                    Orders.Add(new OrderData(10251,4, "Reims", 22.65));
+                    Orders.Add(new OrderData(10252,5, "Lyon", 63.43));
+                    Orders.Add(new OrderData(10253,6, "Charleroi",56.98));
+                    Orders.Add(new OrderData(10254,7, "Rio de Janeiro", 45.65));
+                    Orders.Add(new OrderData(10255,8, "Münster", 11.13));
+                    Orders.Add(new OrderData(10256,9, "Reims", 87.59));
+                    code += 5;
+                }
+            }
+            return Orders;
+        }
+        public int? OrderID { get; set; }
+        public int? EmployeeID { get; set; }
+        public string ShipCity { get; set; }
+        public double? Freight { get; set; }
+    }
+{% endhighlight %}
+{% endtabs %}
+
+{% previewsample "https://blazorplayground.syncfusion.com/embed/VXrAsWDMKuxGenrb?appbar=false&editor=false&result=true&errorlist=false&theme=bootstrap5" %}
+
+## Customize filter UI in foreignkey column
+
+The Syncfusion Grid allows you to customize the filtering user interface (UI) for foreign key columns by using the [FilterTemplate](https://help.syncfusion.com/cr/blazor/Syncfusion.Blazor.Grids.GridColumn.html#Syncfusion_Blazor_Grids_GridColumn_FilterTemplate)  property. By default, a dropdown component is used for filtering foreign key columns. However, you can create your own custom filtering UI by [FilterTemplate](https://help.syncfusion.com/cr/blazor/Syncfusion.Blazor.Grids.GridColumn.html#Syncfusion_Blazor_Grids_GridColumn_FilterTemplate) property. Here’s an example that demonstrates how to create a custom filtering UI in a foreign key column:
+
+> For all filter types other than FilterBar, filtering parameters will be sent in the form of `PredicateModel<T>`. Here, T represents the type of `ForeignKeyValue` property when using the foreignkey column.
+
+In this example, a DropDownList component is rendered as the filter UI for the **“EmployeeID”** foreign key column. The  **DataSource** property of the `SfDropDownList` component is set to the employees data, and the Fields property is configured to display the FirstName field as the text and EmployeeID field as the value. The `value` property is set to the current filter value of the column.
+
+{% tabs %}
+{% highlight razor tabtitle="Index.razor" %}
+@using Syncfusion.Blazor.Grids
+@using Syncfusion.Blazor.DropDowns
+@using BlazorApp1.Data
+
+<SfGrid DataSource="@Orders" Height="315" Toolbar="@(new List<string>() { "Add", "Edit", "Delete", "Cancel", "Update" })">
+    <GridEditSettings AllowAdding="true" AllowDeleting="true" AllowEditing="true"></GridEditSettings>
```

Comment:
change this to Menu filter. check with angular samples

Reply:
modified